BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,234 @@
+             help="Install path for {} qt5 libraries".format(arch),
+             default=qt5_path[arch])
+
+    args = vars(parser.parse_args());
+    dst_app = args["dst_app"]
+    build_target = args["target"]
+    codesign_identity = args["codesign"]
+
+    for arch in architectures:
+        pkg_config_path[arch] = args["{}_pkg_config".format(arch)]
+        qt5_path[arch] = args["{}_qt5_path".format(arch)]
+
+
+def lipo(path0,path1,dst):
+    if subprocess.call(['lipo', '-create', '-output', dst, path0, path1]) != 0:
+        print("WARNING: {} and {} are different but can not be lipo'd, keeping {}".format(path0, path1, path0));
+        shutil.copy(path0, dst)
+
+# Merges two build trees together for different architectures into a single
+# universal binary.
+# The rules for merging are:
+# 1) Files that exist in either src tree are copied into the dst tree
+# 2) Files that exist in both trees and are identical are copied over
+#    unmodified
+# 3) Files that exist in both trees and are non-identical are lipo'd
+# 4) Symlinks are created in the destination tree to mirror the hierarchy in
+#    the source trees
+
+def recursiveMergeBinaries(src0,src1,dst):
+    #loop over all files in src0
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0);
+        newpath1 = os.path.join(src1,filename);
+        new_dst_path = os.path.join(dst,filename);
+        if not os.path.islink(newpath0):
+            if os.path.exists(newpath1):
+                if os.path.isdir(newpath1):
+                    os.mkdir(new_dst_path);
+                    #recurse into directories
+                    recursiveMergeBinaries(newpath0,newpath1,new_dst_path)
+                else:
+                    if filecmp.cmp(newpath0,newpath1):
+                        #copy files that are the same
+                        shutil.copy(newpath0,new_dst_path);
+                    else:
+                        #lipo together files that are different
+                        lipo(newpath0,newpath1,new_dst_path)
+            else:
+              #copy files that don't exist in path1
+              shutil.copy(newpath0,new_dst_path)
+
+    #loop over files in src1 and copy missing things over to dst
+    for newpath1 in glob.glob(src1+"/*"):
+        filename = os.path.basename(newpath0);
+        newpath0 = os.path.join(src0,filename);
+        new_dst_path = os.path.join(dst,filename);
+        if not os.path.exists(newpath0) and not os.path.islink(newpath1):
+            shutil.copytree(newpath1,new_dst_path);
+
+    #fix up symlinks for path0
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0);
+        new_dst_path = os.path.join(dst,filename);
+        if os.path.islink(newpath0):
+            relative_path = os.path.relpath(os.path.realpath(newpath0),src0)
+            os.symlink(relative_path,new_dst_path);
+    #fix up symlinks for path1
+    for newpath1 in glob.glob(src1+"/*"):
```

Comment:
Can't this code be deduplicated using a `for loop`? Or is this check preventing it:
```python
if os.path.islink(newpath1) and not os.path.exists(newpath0):
```

Reply:
I refactored this section to reduce the redundancy. It will be fixed in my next push.